sources/OpenMcdf.Extensions/OLEProperties/PropertyFactory.cs

```diff
@@ -412,26 +412,86 @@ public override string ReadScalarValue(System.IO.BinaryReader br)
                 uint size = br.ReadUInt32();
                 data = br.ReadBytes((int)size);
 
-                return Encoding.GetEncoding(codePage).GetString(data);
+                var result = Encoding.GetEncoding(codePage).GetString(data);
+                //result = result.Trim(new char[] { '\0' });
+
+                //if (this.codePage == CodePages.CP_WINUNICODE)
+                //{
+                //    result = result.Substring(0, result.Length - 2);
+                //}
+                //else
+                //{
+                //    result = result.Substring(0, result.Length - 1);
+                //}
+
+                return result;
             }
 
             public override void WriteScalarValue(BinaryWriter bw, string pValue)
             {
-                data = Encoding.GetEncoding(codePage).GetBytes(pValue);
-                uint dataLength = (uint)data.Length;
+                //bool addNullTerminator = true;
+
+                if (String.IsNullOrEmpty(pValue)) //|| String.IsNullOrEmpty(pValue.Trim(new char[] { '\0' })))
+                {
+                    bw.Write((uint)0);
+                }
+                else if (this.codePage == CodePages.CP_WINUNICODE)
+                {
 
-                // The string data must be null terminated, so add a null byte if there isn't one already
-                bool addNullTerminator =
-                    dataLength == 0 || data[dataLength - 1] != '\0';
+                    data = Encoding.GetEncoding(codePage).GetBytes(pValue);
+
+                    //if (data.Length >= 2 && data[data.Length - 2] == '\0' && data[data.Length - 1] == '\0')
+                    //    addNullTerminator = false;
+
+                    uint dataLength = (uint)data.Length;
+
+                    //if (addNullTerminator)
+                    dataLength += 2;            // null terminator \u+0000
+
+                   // var mod = dataLength % 4;       // pad to multiple of 4 bytes

```

Comment:
Don't unicode strings also have to be padded to 4 bytes?

https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-oleps/9660cb24-953a-4e60-adf2-37cc0e779d19

Reply:
There's another set of padding code up where this is called from (https://github.com/ironfede/openmcdf/blob/c58985f1a7d85f8e128b7f5bd6be371c6b3e5c6d/sources/OpenMcdf.Extensions/OLEProperties/TypedPropertyValue.cs#L132) - so I think the code up their should handle both Lpstr and Lpwstr without needing to have multiple sets of code here.